### ISSUE TITLE
0.7.0 fails start on Ubuntu 18.04, 20.04 due to CoreDNS crash loop

### ISSUE THREAD

**Issue by cameronelliott** [k0sproject/k0s#460]
tldr #2;
k0s fails to start (0.7.0) happy on both pristine Ubuntu 18.04 and 20.04 due to CoreDNS crash looping.

tldr;
I my first shot at k0s was not easy, I ran into the CoreDNS crash loop,
(and only by luck and persistence and time did I find the mention here): [troubleshooting](https://github.com/k0sproject/k0s/blob/main/docs/troubleshooting.md#coredns-in-crashloop)

My first impression was: 
"k0s is not ready for prime-time"

I would have found it helpful to have found an issue to thumbs-up my experience showing
that this was a roadblock for me on my first contact with k0s.

I would suggest there should be an open issue to track thumbs-ups and comments to 
provide feedback to the core team about how prevalent this is and provide feedback about
how to prioritize it.

I almost ran back to k3s after things didn't work from the get-go.

I love the work you are doing.

Please remember the hardest thing on your journey might
be to understand how the new users are struggling with their first contact with k0s.

**Version**
```
0.7.0
```
**Platform**
Which platform did you run k0s on?
```
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 18.04.4 LTS
Release:	18.04
Codename:	bionic
```
**What happened?**
I am new to k0s, I tried to do the vanilla install described here: https://www.mirantis.com/blog/how-to-set-up-k0s-kubernetes-a-quick-and-dirty-guide/

**How To Reproduce**
Not quite sure how my Ubuntu 18.04 differs from a vanilla 18.04,
but the team already seems aware of the coredns issue.

**Expected behavior**
Stable launch of k0s on my first try. :)

**Comment by cameronelliott** [comment]
Actually, still trying to fix this, I know this is really a CoreDNS issue/kubernetes issue (not k0s directly), but for the k8s newbies trying out k0s, it is a show stopper.

I am trying the first approach mentioned by CoreDNS to fix this issue:
```
Add the following to your kubelet config yaml: resolvConf: <path-to-your-real-resolv-conf-file> (or via command line flag --resolv-conf deprecated in 1.10). Your “real” resolv.conf is the one that contains the actual IPs of your upstream servers, and no local/loopback address. This flag tells kubelet to pass an alternate resolv.conf to Pods. For systems using systemd-resolved, /run/systemd/resolve/resolv.conf is typically the location of the “real” resolv.conf, although this can be different depending on your distribution.
```
But I am super k8s inexperienced.
I tried adding `resolvConf: /run/systemd/resolve/resolv.conf` in the file `/var/lib/k0s/kubelet-config.yaml`
But the changes I make keep getting removed/overwritten.
*Is `kubelet-config.yaml` generated somehow from k0s.yaml ?*
*Does the re-writing of kubelet-config.yaml indicate: a) generation from other source b) an error in the yaml ??*
*Where would I put `resolvConf: /run/systemd/resolve/resolv.conf` if NOT in the  file `/var/lib/k0s/kubelet-config.yaml`??*

Thanks for any help.
I hope my k8s/k0s new eyes can provide helpful feedback.

**Comment by cameronelliott** [comment]
Also, if anyone would care to explain how to test CoreDNS without running `sudo k0s server --enable-worker`
each time to see if it is fixed, that would be super helpful.
Thanks

**Comment by cameronelliott** [comment]
You know, if I become a serious maintainer, I have to learn to read issue comments from last-to-first. :)

The first thing I tried:
I had tried installing k0s on Ubuntu 18.04, on a system I have that is a couple years old, and it hit the CoreDNS loop issue,
but I can't say that system is in a pristine state, so....

After a lot of unsuccessful work, I tried creating a pristine Ubuntu 18.04 LTS at Digital Ocean,
and surprisingly (to me), I am having the same CoreDNS fail issue on a clean/pristine ubuntu 18.04 system:
`kube-system   coredns-5c98d7d4d8-p4hvx                   0/1     CrashLoopBackOff   8          13m`

I haven't quite figured out how to make CoreDNS happy on Ubuntu yet.

**Comment by cameronelliott** [comment]
I also tried installing k0s on a Digital Ocean 20.04 instance, and everything fails due to CoreDNS crash loops.

**Comment by trawler** [comment]
@cameronelliott thank you for this report. We actually already implemented a fix for the above mentioned problem, but perhaps you are running into a different issue... please provide:
* What is the command you are using to run k0s?
* What are the flags you see for your kubelet instance? (you should have a `--resolv-conf` with the value `/run/systemd/resolve/resolv.conf`

Thanks!

**Comment by trawler** [comment]
> @cameronelliott thank you for this report. We actually already implemented a fix for the above mentioned problem, but perhaps you are running into a different issue... please provide:
> 
> * What is the command you are using to run k0s?
> * What are the flags you see for your kubelet instance? (`ps aux | grep kubelet`). You should have a `--resolv-conf` flag with the value `/run/systemd/resolve/resolv.conf`
> 
> Thanks!

Oops, my mistake. The above mentioned PR will be part of `v0.8.0`: https://github.com/k0sproject/k0s/pull/395
You can try the master version instead, or wait a couple more days for `v0.8.0` to come out.
